app/namespaces/test_adjustment/test_adjustment_service.py

```diff
@@ -0,0 +1,267 @@
+    y_se = fields.Float()
+    n_sp = fields.Integer()
+    y_sp = fields.Float()
+
+
+class TestAdjHandler:
+    def __init__(self, model_code=testadj_model_code, model_name='testadj_binomial_se_sp',
+                 execution_params={}):
+        self.TESTADJ_MODEL = self.get_stan_model_cache(model_code=model_code, model_name=model_name)
+        self.TESTADJ_MODEL_NAME = model_name
+        # Parse execution params
+        self.n_iter = execution_params.get('n_iter', 4000)
+        self.n_chains = execution_params.get('n_chains', 3)
+        self.return_fit = execution_params.get('return_fit', False)
+        self.n_replicates = execution_params.get('n_replicates', 5)
+        self.trials_lim = execution_params.get('trials_lim', 100)
+        self.modelsets_lim = execution_params.get('modelsets_lim', 3)
+
+    # make sure to gitignore model caches, because the model needs to be compiled separately
+    # on each machine - it is system-specific C++ code
+    def get_stan_model_cache(self, model_code: str, model_name: str = 'anon_model', **kwargs: Dict) -> pystan.StanModel:
+        """Use just as you would `pystan.StanModel`"""
+
+        # Create filepath of cached model
+        code_hash = md5(model_code.encode('ascii')).hexdigest()
+        cache_fn = f'stanmodelcache-{model_name}-{code_hash}.pkl'
+
+        # Try to load cached model
+        try:
+            cached_model = pickle.load(open(cache_fn, 'rb'))
+            print(f"Using cached StanModel at filepath {cache_fn}")
+
+        # Otherwise create the model and cache it
+        except FileNotFoundError:
+            cached_model = pystan.StanModel(model_code=model_code, model_name=model_name, **kwargs)
+            with open(cache_fn, 'wb') as f:
+                pickle.dump(cached_model, f)
+        return cached_model
+
+    def fit_one_pystan_model(self, model_params: Dict) -> Tuple:
+        if model_params['n_sp' < 10] or model_params['n_se' < 10]  \
+                or model_params['y_sp' < 5]  or model_params['y_se' < 5]:
+            adapt_delta = 0.99
+        else:
+            adapt_delta = 0.8
+        fit = self.TESTADJ_MODEL.sampling(data=model_params,
+                                          iter=self.n_iter,
+                                          chains=self.n_chains,
+                                          control={'adapt_delta': adapt_delta},
+                                          check_hmc_diagnostics=False)
+
+        summary = fit.summary()
+        summary_df = pd.DataFrame(data=summary['summary'],
+                                  index=summary['summary_rownames'],
+                                  columns=summary['summary_colnames'])
+
+        samples = fit.extract(pars='prev', permuted=True)['prev']
+
+        summary_df_parsed = summary_df.loc['prev', ['50%', '2.5%', '97.5%']]
+        summary_df_parsed['samples'] = samples
+        summary_df_parsed['fit'] = fit
+
+        diagnostics = pystan.diagnostics.check_hmc_diagnostics(fit)
```

Comment:
This is what I was talking about when I said we don't need to check all of the diagnostics' values. We run the HMC diagnostics in line 117, which is great, but then we check to see whether all of them are true on line 119. We'd like to modify this such that we only check whether a couple of these are true.

My guess is that diagnostics is a dict, with keys that include something like 'rhat' and 'n_eff'.

- It might give you boolean values for both rhat and n_eff. If so, make sure that they're both true.
- It might give you numeric values for both. If so, make sure that `0.9 < Rhat < 1.1` and that `n_eff > 500`